src/main/java/mate/academy/bookstore/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+
+    @Override
+    @Transactional
+    public List<OrderDto> getOrders(Long userId) {
+        List<Order> orders = orderRepository.findByUserId(userId)
+                .orElseThrow(() -> new EntityNotFoundException("No orders found for user with id "
+                        + userId));
+
+        return orderMapper.map(orders);
+    }
+
+    @Override
+    @Transactional
+    public OrderDto createOrder(Long userId, CreateOrderRequestDto createOrderRequestDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId)
+                .orElseThrow(() -> new EntityNotFoundException("Shopping cart not found "
+                        + "for user id " + userId));
+
+        BigDecimal total = calculateTotal(shoppingCart);
+
+        Order order = createNewOrder(shoppingCart, total,
+                createOrderRequestDto.getShippingAddress());
+
+        createAndSaveOrderItems(shoppingCart, order);
+
+        clearShoppingCart(shoppingCart);
+
+        OrderDto orderDto = orderMapper.toDto(order);
+        Set<OrderItemDto> orderItemDtos = getOrderItems(order.getId());
+        orderDto.setOrderItems(orderItemDtos);
+        return orderDto;
+    }
+
+    private Order createNewOrder(ShoppingCart shoppingCart, BigDecimal total,
+                                 String shippingAddress) {
+        Order order = new Order();
+        order.setUser(shoppingCart.getUser());
+        order.setStatus(Order.Status.PENDING);
+        order.setTotal(total);
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(shippingAddress);
+        return orderRepository.save(order);
+    }
+
+    private void createAndSaveOrderItems(ShoppingCart shoppingCart, Order order) {
+        shoppingCart.getCartItems().forEach(cartItem -> {
+            OrderItem orderItem = new OrderItem();
+            orderItem.setOrder(order);
+            orderItem.setBook(cartItem.getBook());
+            orderItem.setQuantity(cartItem.getQuantity());
+            orderItem.setPrice(cartItem.getBook().getPrice());
+            orderItemRepository.save(orderItem);
+        });
+    }
+
+    private void clearShoppingCart(ShoppingCart shoppingCart) {
+        shoppingCart.getCartItems().clear();
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    @Transactional
+    public OrderDto updateOrderStatus(Long orderId, Order.Status status) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException("Order not found"));
```

Comment:
include relevant details in exception messages to facilitate debugging (like ID here)